sdk/containerregistry/azure-containers-containerregistry/src/main/java/com/azure/containers/containerregistry/implementation/authentication/ContainerRegistryTokenCredentialImpl.java

```diff
@@ -0,0 +1,43 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+package com.azure.containers.containerregistry.implementation.authentication;
+
+import com.azure.core.credential.AccessToken;
+import com.azure.core.credential.TokenCredential;
+import com.azure.core.credential.TokenRequestContext;
+import reactor.core.publisher.Mono;
+
+/**
+ * Token credentials representing the container registry refresh token.
+ * This token is unique per registry operation.
+ */
+class ContainerRegistryTokenCredentialImpl {
```

Comment:
As discussed offline, will this type be implementing `TokenCredential`?

Reply:
it may. As of now since the surface is non-public, we have the flexibility of changing it in future.

---
In reply to: [597957584](https://github.com/Azure/azure-sdk-for-java/pull/19926#discussion_r597957584) [](ancestors = 597957584)